packages/lit-html/src/directives/cache.ts

```diff
@@ -45,17 +58,23 @@ class CacheDirective extends Directive {
     // into the cache.
     if (
       isTemplateResult(this._value) &&
-      (!isTemplateResult(v) || this._value.strings !== v.strings)
+      (!isTemplateResult(v) ||
+        getStringsFromTemplate(this._value) !== getStringsFromTemplate(v))
     ) {
       // This is always an array because we return [v] in render()
       const partValue = getCommittedValue(containerPart) as Array<ChildPart>;
       const childPart = partValue.pop()!;
-      let cachedContainerPart = this._templateCache.get(this._value.strings);
+      let cachedContainerPart = this._templateCache.get(
+        getStringsFromTemplate(this._value)
+      );
       if (cachedContainerPart === undefined) {
         const fragment = document.createDocumentFragment();
         cachedContainerPart = render(nothing, fragment);
         cachedContainerPart.setConnected(false);
-        this._templateCache.set(this._value.strings, cachedContainerPart);
+        this._templateCache.set(
+          getStringsFromTemplate(this._value),
```

Comment:
You call this several times. Can you put the result in a variable?

Reply:
Done. Lifted the calls to the top.